### PR TITLE
Server: Add support for shipping traces to an otel collector

### PR DESCRIPTION
**Pull request by douglascamata**
## What's done in the PR:

* Add support for telemeter/rhelemeter servers to ship traces in OTEL format to an OTEL collector.
  * Changes their integration tests to run an OTEL collector and ship traces to it.
* Log request details using trace and span ids in both Telemeter (used a randomly generated request id before) and Rhelemeter.
  * We don't need to generate yet another request identifier if we already have the trace and span ids there. The proposal is to standardize around the traces.

## Took some extra time to...

* ~~Upgrade Prometheus used in the integration tests from a release old enough that it didn't have a build for my OS/architecture to the most recent.~~
  * Gave up on this: it triggers failures in integration tests that are very difficult to fix due to a chain of dependency upgrades that are required.

## Verification

Verified it works by running the integration tests with Rhelemeter on debug level and a local OTEL collector receiving the traces.

### CODE REVIEW

**Comment by douglascamata** [comment]
Seeing some weird CI failures and trying to investigate:

```
error: failed to push image registry.build01.ci.openshift.org/ci-op-lf1nky9g/release:latest: unable to upload new layer (0): Patch "https://registry.build01.ci.openshift.org/v2/ci-op-lf1nky9g/release/blobs/uploads/1dd4cd97-9b41-4af5-9227-96247ddba9c6?_state=VPkkGVZBjSd6-JxdpBszU2B6HnAClvm_D3B4wuTrO-F7Ik5hbWUiOiJjaS1vcC1sZjFua3k5Zy9yZWxlYXNlIiwiVVVJRCI6IjFkZDRjZDk3LTliNDEtNGFmNS05MjI3LTk2MjQ3ZGRiYTljNiIsIk9mZnNldCI6MCwiU3RhcnRlZEF0IjoiMjAyMy0xMC0wOVQwOTo0MDoxMy43MzEyNjU4MjlaIn0%3D": operator "cluster-network-operator" contained an invalid image-references file: no input image tag named "kuryr-cni"
{"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"k8s.io/test-infra/prow/entrypoint/run.go:84","func":"k8s.io/test-infra/prow/entrypoint.Options.internalRun","level":"error","msg":"Error executing test process","severity":"error","time":"2023-10-09T09:40:14Z"} 
```

**Comment by douglascamata** [comment]
Found the reason and solution to the build failures in https://github.com/openshift/cluster-network-operator/pull/2056. Waiting for it to be merged.

File: cmd/rhelemeter-server/main.go [hunk 1]
```diff
@@ -170,6 +172,7 @@ type Options struct {
 	TracingEndpoint         string
 	TracingEndpointType     string
 	TracingSamplingFraction float64
+	TracingInsecure         bool
```
Comment (philipgough) [inline cmd/rhelemeter-server/main.go:175]: Whats the purpose of this? I would expect the server to push traces into some collector and/or sidecar but perhaps there is something we need to expose?

**Review by philipgough — APPROVED** [review]
/lgtm

**Review by moadz — COMMENTED** [review]
`vendor` files should be `.gitignore`'ed

**Pull request merged**